pkg/asset/agent/manifests/nmstateconfig.go

```diff
@@ -242,41 +243,68 @@ func (n *NMStateConfig) validateNMStateLabels() field.ErrorList {
 	return allErrs
 }
 
-func getFirstIP(nmStateConfig *nmStateConfig) string {
+func getFirstIP(nmstateRaw []byte) (string, error) {
+	var nmStateConfig nmStateConfig
+	err := yaml.Unmarshal(nmstateRaw, &nmStateConfig)
+	if err != nil {
+		return "", fmt.Errorf("error unmarshalling NMStateConfig: %w", err)
+	}
+
 	for _, intf := range nmStateConfig.Interfaces {
 		for _, addr4 := range intf.IPV4.Address {
 			if addr4.IP != "" {
-				return addr4.IP
+				return addr4.IP, nil
 			}
 		}
 		for _, addr6 := range intf.IPV6.Address {
 			if addr6.IP != "" {
-				return addr6.IP
+				return addr6.IP, nil
 			}
 		}
 	}
-	return ""
+
+	return "", nil
 }
 
-// GetNodeZeroIP retrieves the first IP from the user provided NMStateConfigs to set as the node0 IP
-func GetNodeZeroIP(nmStateConfigs []*aiv1beta1.NMStateConfig) (string, error) {
-	for i := range nmStateConfigs {
-		var nmStateConfig nmStateConfig
-		err := yaml.Unmarshal(nmStateConfigs[i].Spec.NetConfig.Raw, &nmStateConfig)
+// GetNodeZeroIP retrieves the first IP to be set as the node0 IP.
+// The method prioritizes the search by trying to scan first the NMState configs defined
+// in the agent-config hosts - so that it would be possible to skip the worker nodes - and then
+// the NMStateConfig.
+func GetNodeZeroIP(agentConfig *agenttypes.Config, nmStateConfigs []*aiv1beta1.NMStateConfig) (string, error) {
+	rawConfigs := []aiv1beta1.RawNetConfig{}
+
+	// Select first the configs from the hosts, if defined
+	if agentConfig != nil {
+		for _, host := range agentConfig.Hosts {
+			// Skip worker hosts
+			if host.Role == "worker" {
```

Comment:
This is not enough, because say you have a 3+2 HA cluster and the assigned roles are: `"", "", "master", "master", "master"`. In this case the first two hosts _must_ be workers, but we will still select the IP from the first one.

Reply:
Here I'd like to avoid here complex logic as we did in the `validateHostCount` (which I originally implemented in the baremetal part), in favor of a simpler best effort approach, oriented to promote the explicit configuration when available. 
We could change the implementation to include only those hosts having the `role` field equal to `master` (thus ignoring the `workers` and the ones not defined) - and it's fine for me, even though there would be always a partial configuration (the original sin) setup that wouldn't be fine (`"", "", "worker", "worker", "worker"`).
It'd be anyhow something easy to explain in the documentation, ie when the rendezvousIP is not defined then it will be choose one of the static IP configured among the master hosts only.

Reply:
I also think you should only include those hosts having role equal to `master` (ignoring `workers` and those not defined)

Reply:
Done

Reply:
> I'd like to avoid here complex logic

The complexity doesn't go away, the question is whether we handle it or push it onto the user.